test_scripts/ATF_ShutDown_IGNITION_OFF.lua

```diff
@@ -0,0 +1,71 @@
+-- Requirement summary:
+-- [Data Resumption]:OnExitAllApplications(IGNITION_OFF) in terms of resumption 
+--
+-- Description:
+-- 1. HMI is running
+--    One App is registered and activated on HMI
+-- 2. Perform ignition Off
+--    HMI sends OnExitAllApplications(IGNITION_OFF)
+--
+-- Expected result:
+-- 1. SDL sends to App OnAppInterfaceUnregistered
+-- 2. SDL sends to HMI OnSDLClose and stops working
+
+
+--[[ Required Shared libraries ]]
+local commonFunctions = require('user_modules/shared_testcases/commonFunctions')
+local commonSteps = require('user_modules/shared_testcases/commonSteps')
+
+--[[ General Settings for configuration ]]
+Test = require('user_modules/dummy_connecttest')
+require('cardinalities')
+require('user_modules/AppTypes')
+
+--[[ Preconditions ]]
+commonFunctions:newTestCasesGroup("Preconditions")
+commonSteps:DeletePolicyTable()
+commonSteps:DeleteLogsFiles()
+
+function Test:Start_SDL_With_One_Activated_App()
+  self:runSDL()
+  commonFunctions:waitForSDLStart(self):Do(function()
+    self:initHMI():Do(function()
```

Comment:
@ttdong  the main benefit and idea imering app preconditions to one test case. 
In my opinion there should be 2 test cases for each test script

- Precondition ( with describing state of precondition in test case name)
- Test case (with description test case crux in name)
- PostCondition 

Also note that in new approach in case if some part preconditions won be succeed, the other ones won't be continued. I that case it is much more easy to analyze the problem. 

Also note, that you do not need to add any implicit generation of test cases with such approach. Everything is explicit. As I told before many times : **Auto generation of test cases is evil**